cedar-drt/fuzz/src/lib.rs

```diff
@@ -33,180 +31,158 @@ use cedar_policy_core::extensions::Extensions;
-                return true
-            }
-            Err(_) => None,
-        };
-        self.def_engine.eval(r, entities, expr, v)
+    let rust_res_for_comparison: cedar_policy::Response = Response {
+        diagnostics: Diagnostics {
+            errors: Vec::new(),
+            ..rust_res.diagnostics
+        },
+        ..rust_res
     }
+    .into();
+    assert_eq!(
+        InterfaceResponse::from(rust_res_for_comparison),
+        definitional_res,
+        "Mismatch for {request}\nPolicies:\n{}\nEntities:\n{}",
+        &policies,
+        &entities
+    );
+    ret
+}
 
-    /// Differentially test the given authorization request.
-    /// Panics if the two engines do not agree.
-    /// Returns the response which the engines agree on.
-    pub fn run_single_test(
-        &self,
-        q: &ast::Request,
-        policies: &PolicySet,
-        entities: &Entities,
-    ) -> Response {
-        let (rust_res, rust_auth_dur) =
-            time_function(|| self.authorizer.is_authorized(q, policies, entities));
-        info!("{}{}", RUST_AUTH_MSG, rust_auth_dur.as_nanos());
-
-        // For now, we ignore all tests where the Rust side returns an integer
-        // overflow error, as the behavior between Rust and Dafny is
-        // intentionally different
-        if rust_res
-            .diagnostics
-            .errors
-            .iter()
-            .any(|e| e.to_string().contains("integer overflow"))
-        {
-            return rust_res;
-        }
-
-        // very important that we return the Rust response, with its rich errors,
-        // in case the caller wants to expect those. (and not the definitional
-        // response, which as of this writing contains less-rich errors)
-        let ret = rust_res.clone();
-
-        let definitional_res = self.def_engine.is_authorized(q, policies, entities);
-        // for now, we expect never to receive errors from the definitional engine,
-        // and we otherwise ignore errors in the comparison
-        assert_eq!(
-            definitional_res
-                .diagnostics()
-                .errors()
-                .map(|e| e.to_string())
-                .collect::<Vec<String>>(),
-            Vec::<String>::new()
-        );
+/// Compare the behavior of the validator in cedar-policy against a custom Cedar
+/// implementation. Panics if the two do not agree.
+pub fn run_val_test(
+    custom_impl: &dyn CedarTestImplementation,
+    schema: ValidatorSchema,
+    policies: &PolicySet,
+    mode: ValidationMode,
+) {
+    let validator = Validator::new(schema.clone());
+    let (rust_res, rust_validation_dur) = time_function(|| validator.validate(policies, mode));
+    info!("{}{}", RUST_VALIDATION_MSG, rust_validation_dur.as_nanos());
+
+    let definitional_res = custom_impl.validate(schema.clone(), policies, mode);
+
+    assert!(
+        definitional_res.parsing_succeeded(),
+        "Dafny json parsing failed for:\nPolicies:\n{}\nSchema:\n{:?}",
```

Comment:
Error message probably shouldn't refer to Dafny, since this function is generic over all `CedarTestImplementation`s